### PR TITLE
vxlan: add windows support

### PR DESCRIPTION
**Pull request by rakelkar**
## Description
VXLAN support for windows (3b from #832) #833
Needs Integration Tests!!

## Todos
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: backend/vxlan/vxlan_network_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// +build windows
+
+// Copyright 2015 CNI authors
```
Comment (tomdee) [inline backend/vxlan/vxlan_network_windows_test.go]: Wrong authors (and year) :smile:

**Comment by anikundesu** [comment]
@rakelkar
The reason why Travis CI build failed is lack of "EoF" at the last of backend/vxlan/vxlan_network_windows.go and backend/vxlan/vxlan_network_windows_test.go as linux-based text files.
To fix the failure, these two files should be opened on the linux-based system, add one line at the last of the file, and delete the last line. I used vi to do it and successfully executed "make test" on a linux-based system.

------
Takashi Kanai
Microsoft MVP of Cloud and Datacenter Management

File: backend/vxlan/vxlan_network_windows_test.go (hunk 1 above)
Comment (madhanrm) [reply backend/vxlan/vxlan_network_windows_test.go]: Resolved.

**Comment by rakelkar** [comment]
@anikundesu nice find! Could open a PR into the source branch with your fix?

**Comment by anikundesu** [comment]
@rakelkar The fix I pointed out has already been commited by @madhanrm at dd0cd9e

File: backend/vxlan/vxlan_windows.go [hunk 2]
```diff
@@ -11,14 +13,181 @@
...
+		return nil, err
+
+	default:
+		return nil, fmt.Errorf("failed to acquire lease: %v", err)
+	}
+
+	// check if the network exists and has the expected settings?
+	networkName := cfg.name
+	createNetwork := true
+	addressPrefix := config.Network
```
Comment (alekssaul) [inline backend/vxlan/vxlan_windows.go]: is this not supposed to be
```
	addressPrefix := l.Subnet
	networkGatewayAddress := l.Subnet.IP + 1
```
lease information is acquired but not used during HNSNetwork creation afaik I can understand?

Comment (madhanrm) [reply backend/vxlan/vxlan_windows.go]: @alekssaul  Yes, you are correct

**Comment by tomdee** [comment]
@rakelkar What is the status of this PR? Is it ready for review and merging?

**Comment by madhanrm** [comment]
@tomdee, there also seems to be another PR : #1042 which is doing similar to this PR. Can we get this PR merged & then work on top if it. The binaries out of this PR has been tested for over an year now

**Comment by dineshgovindasamy** [comment]
Lets abandon this PR. We are taking PR#1042 which is merged PR.

Rakesh, can u abandon this PR?

**Comment by dineshgovindasamy** [comment]
tagging @rakelkar

**Comment by rajatchopra** [comment]
Closing this PR in favour of #1042. Please re-open if needed.

**Pull request closed without merging**